presto-main/src/main/java/com/facebook/presto/server/TaskResource.java

```diff
@@ -311,6 +333,34 @@ public void getResults(
         requireNonNull(taskId, "taskId is null");
         requireNonNull(bufferId, "bufferId is null");
 
+        if (shutdownHandler.getNoTaskAtGracefulShutdown().get()) {
```

Comment:
Starting to loose track of the code around here :-) So I was wondering if some of this PR has any fragments from the "fast draining" PR ?